Add documentation for custom assertions (#129)
pester/docs: docs/assertions/assertions.mdx

$true | Should -Not -Be $false
```

### Because

Lets you to add an optional message to the default test error message to tell the user why the test should pass.

```powershell

> $actual = "Actual value"

Reviewer: Maybe reuse the Get-Planet example with `-Name M*` which should return 'Mercur, Mars' -Because "those are the only two planets starting with M*'?

Author: Updated example

2 | Should -BeGreaterOrEqual 2
```

### BeGreaterThan

Asserts that a number is greater than an expected value. Uses PowerShell's -gt operator to compare the two values.

```powershell
$Error.Count | Should -BeGreaterThan 0
[version]'2.3.4.5' | Should -BeGreaterThan ([version]'1.0.0.1')
(Get-Date).AddMinutes(5) | Should -BeGreaterThan (Get-Date)
```

### BeIn

> Asserts that the actual value is contained by the array/collection

Reviewer: ```suggestion
Asserts that the actual value is present in the array/collection
```

$actual | Should -BeLikeExactly "actual *" # Test will fail
```

### BeNullOrEmpty

Checks values for null or empty (strings). The static [String]::IsNullOrEmpty() method is used to do the comparison.

```powershell

> $null | Should -BeNullOrEmpty # Test will pass
> $null | Should -Not -BeNullOrEmpty # Test will fail
> @()   | Should -BeNullOrEmpty # Test will pass
> ""    | Should -BeNullOrEmpty # Test will pass

Reviewer: ```suggestion
$null | Should -BeNullOrEmpty # Test will pass
@()   | Should -BeNullOrEmpty # Test will pass
""    | Should -BeNullOrEmpty # Test will pass

$null | Should -Not -BeNullOrEmpty # Test will fail
```

Reviewer: Seems less confusing this way.
